assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -80,6 +80,120 @@ describe( 'CurrentSurvey', () => {
 		expect( container ).toMatchSnapshot();
 	} );
 
+	it( "should render an open text question when the `question_type` is 'open_text'", async () => {
+		registry
+			.dispatch( CORE_USER )
+			.receiveTriggerSurvey( fixtures.singleQuestionOpenText, {
+				triggerID: 'jestSurvey',
+			} );
+
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/user\/data\/survey-event/,
+			{ body: {}, status: 200 }
+		);
+
+		const {
+			getByText,
+			getByPlaceholderText,
+			getByRole,
+			getByDisplayValue,
+			findByText,
+		} = render( <CurrentSurvey />, {
+			registry,
+		} );
+
+		expect( fetchMock ).toHaveFetched(
+			/^\/google-site-kit\/v1\/core\/user\/data\/survey-event/
+		);
+
+		expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+		// Submit button should be disabled if text input is empty
+		expect( getByRole( 'button', { name: 'Submit' } ) ).toHaveAttribute(
+			'disabled'
+		);
+
+		// Check question_text is set
+		expect(
+			getByText( 'How satisfied are you with Site Kit?' )
+		).toBeInTheDocument();
+
+		// Check subtitle is set
+		expect(
+			getByText( 'Based on your experience so far, tell us.' )
+		).toBeInTheDocument();
+
+		const STRING_100_CHARACTERS =
+			'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit auctor dui, id faucibus nisl';
+
+		const STRING_110_CHARACTERS = STRING_100_CHARACTERS + ' rhoncus n';
+
+		// Now enter answer text
+		fireEvent.change( getByPlaceholderText( 'Write here' ), {
+			target: { value: STRING_110_CHARACTERS },
+		} );
+
+		// Input does not have a label so this is the best way to assert the text has been entered correctly
+		expect(
+			getByDisplayValue( STRING_100_CHARACTERS )
+		).toBeInTheDocument();
+
+		// Submit button should be enabled if text has been entered
+		expect( getByRole( 'button', { name: 'Submit' } ) ).not.toHaveAttribute(
+			'disabled'
+		);
+
+		// check clearing and entering input again
```

Comment:
These are super useful, thanks 🙂 

Nitpick, but can you make all of these comments full sentences including ending with a fullstop?

Reply:
Done.